pkg/tfgen/convert_cli.go

```diff
@@ -0,0 +1,427 @@
+	mappings []tfbridge.ProviderInfo,
+) (
+	output map[string]translatedExample,
+	finalError error,
+) {
+	outDir, err := os.MkdirTemp("", "bridge-examples-output")
+	if err != nil {
+		finalError = fmt.Errorf("convertViaPulumiCLI: failed to create a temp dir "+
+			" bridge-examples-output: %w", err)
+		return
+	}
+	defer func() {
+		if err := os.RemoveAll(outDir); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples-output dir: %w", err)
+			}
+		}
+	}()
+
+	examplesJSON, err := os.CreateTemp("", "bridge-examples.json")
+	if err != nil {
+		finalError = fmt.Errorf("convertViaPulumiCLI: failed to create a temp "+
+			" bridge-examples.json file: %w", err)
+		return
+	}
+	defer func() {
+		if err := os.Remove(examplesJSON.Name()); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples.json file: %w", err)
+			}
+		}
+	}()
+
+	// Write example to bridge-examples.json.
+	examplesBytes, err := json.Marshal(examples)
+	if err != nil {
+		finalError = fmt.Errorf("convertViaPulumiCLI: failed to marshal examples "+
+			"to JSON: %w", err)
+		return
+	}
+	if err := os.WriteFile(examplesJSON.Name(), examplesBytes, 0655); err != nil {
+		finalError = fmt.Errorf("convertViaPulumiCLI: failed to write a temp "+
+			"bridge-examples.json file: %w", err)
+		return
+	}
+
+	mappingsDir := filepath.Join(outDir, "mappings")
+
+	mappingsFile := func(i int, p tfbridge.ProviderInfo) string {
+		return filepath.Join(mappingsDir, fmt.Sprintf("%s.json", p.Name))
+	}
+
+	// Write out mappings files if necessary.
+	for i, m := range mappings {
+		if i == 0 {
+			if err := os.MkdirAll(mappingsDir, 0755); err != nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to write "+
+					"mappings folder: %w", err)
+				return
+			}
+		}
+		mpi := tfbridge.MarshalProviderInfo(&m)
+		bytes, err := json.Marshal(mpi)
+		if err != nil {
+			finalError = fmt.Errorf("convertViaPulumiCLI: failed to write "+
+				"mappings folder: %w", err)
+			return
+		}
+		mf := mappingsFile(i, m)
+		if err := os.WriteFile(mf, bytes, 0655); err != nil {
+			finalError = fmt.Errorf("convertViaPulumiCLI: failed to write "+
+				"mappings file: %w", err)
+			return
+		}
+	}
+
+	pulumiPath, err := exec.LookPath("pulumi")
+	if err != nil {
+		finalError = fmt.Errorf("convertViaPulumiCLI: pulumi executalbe not "+
+			"in PATH: %w", err)
+		return
+	}
+
+	var mappingsArgs []string
+	for i, m := range mappings {
+		mappingsArgs = append(mappingsArgs, "--mappings", mappingsFile(i, m))
+	}
+
+	cmd := exec.Command(pulumiPath, append([]string{"convert",
+		"--from", "terraform",
+		"--language", "pcl",
+		"--out", outDir,
+		"--generate-only",
+	}, mappingsArgs...)...)
+
+	cmd.Dir = filepath.Dir(examplesJSON.Name())
+	cmd.Env = append(os.Environ(),
+		"PULUMI_CONVERT_EXAMPLES="+filepath.Base(examplesJSON.Name()))
```

Comment:
Had to use an ENV var since the --convert-examples arg actually fails to parse.

Reply:
You we're probably missing a double dash:
`pulumi convert --from terraform --lanugage pal -- --convert-examples examples.json`
Note the -- before --convert-examples.

Reply:
I've tried that and it didn't work. I can try again.

Reply:
Nope @Frassle that makes the test fail. There's no error from pulumi but it doesn't generate the output file.